src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,444 @@
+            m_ui->checkDisablePEX->setChecked(isFirstTorrentPEXDisabled);
+        if (m_ui->checkDisableLSD->checkState() != Qt::PartiallyChecked)
+            m_ui->checkDisableLSD->setChecked(isFirstTorrentLSDDisabled);
+    }
+    else
+    {
+        m_ui->checkDisableDHT->setChecked(true);
+        m_ui->checkDisableDHT->setEnabled(false);
+        m_ui->checkDisablePEX->setChecked(true);
+        m_ui->checkDisablePEX->setEnabled(false);
+        m_ui->checkDisableLSD->setChecked(true);
+        m_ui->checkDisableLSD->setEnabled(false);
+    }
+    const QString privateTorrentsTooltip = tr("Not applicable to private torrents");
+    m_ui->checkDisableDHT->setToolTip(privateTorrentsTooltip);
+    m_ui->checkDisablePEX->setToolTip(privateTorrentsTooltip);
+    m_ui->checkDisableLSD->setToolTip(privateTorrentsTooltip);
+
+    initialValues =
+    {
+        m_ui->spinUploadLimit->value(),
+        m_ui->spinDownloadLimit->value(),
+        getRatio(),
+        getSeedingTime(),
+        m_ui->checkDisableDHT->checkState(),
+        m_ui->checkDisablePEX->checkState(),
+        m_ui->checkDisableLSD->checkState()
+    };
+
+    // Sync up/down speed limit sliders with their corresponding spinboxes
+    connect(m_ui->sliderUploadLimit, &QSlider::valueChanged, this, &TorrentOptionsDialog::updateUploadSpinValue);
+    connect(m_ui->sliderDownloadLimit, &QSlider::valueChanged, this, &TorrentOptionsDialog::updateDownloadSpinValue);
+    connect(m_ui->spinUploadLimit, qOverload<int>(&QSpinBox::valueChanged)
+            , this, &TorrentOptionsDialog::updateUploadSliderValue);
+    connect(m_ui->spinDownloadLimit, qOverload<int>(&QSpinBox::valueChanged)
+            , this, &TorrentOptionsDialog::updateDownloadSliderValue);
+
+    connect(m_ui->checkMaxRatio, &QCheckBox::toggled, this
+            , [this]() { m_ui->spinRatioLimit->setEnabled(m_ui->checkMaxRatio->isChecked()); });
+    connect(m_ui->checkMaxTime, &QCheckBox::toggled, this
+            , [this]() { m_ui->spinTimeLimit->setEnabled(m_ui->checkMaxTime->isChecked()); });
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    connect(m_ui->buttonGroup, &QButtonGroup::idClicked, this, &TorrentOptionsDialog::handleRatioTypeChanged);
+#else
+    connect(m_ui->buttonGroup, qOverload<int>(&QButtonGroup::buttonClicked)
+            , this, &TorrentOptionsDialog::handleRatioTypeChanged);
+#endif
+
+    Utils::Gui::resize(this);
+}
+
+TorrentOptionsDialog::~TorrentOptionsDialog()
+{
+    delete m_ui;
+}
+
+void TorrentOptionsDialog::accept()
+{
+    if (m_ui->radioTorrentLimit->isChecked() && !m_ui->checkMaxRatio->isChecked() && !m_ui->checkMaxTime->isChecked())
+    {
+        QMessageBox::critical(this, tr("No share limit method selected"), tr("Please select a limit method first"));
+        return;
+    }
+
+    const auto *session = BitTorrent::Session::instance();
+    for (const BitTorrent::InfoHash &hash : m_torrentHashes)
```

Comment:
`for (const BitTorrent::InfoHash &hash : asConst(m_torrentHashes()`

Reply:
Forgot about that, thanks.